internal_coroutines_test.go

```diff
@@ -1034,3 +1034,53 @@ func TestDecodeFutureChain(t *testing.T) {
 	}
 	require.EqualValues(t, expected, history)
 }
+
+func TestSelectFuture_WithBatchSets(t *testing.T) {
+	var history []string
+	d := newDispatcher(background, func(ctx Context) {
+		future1, settable1 := NewFuture(ctx)
+		future2, settable2 := NewFuture(ctx)
+		future3, settable3 := NewFuture(ctx)
+
+		s := NewSelector(ctx)
```

Comment:
For go's select: If one or more of the communications can proceed, a single one that can proceed is chosen via a uniform pseudo-random selection.
https://golang.org/ref/spec#Select_statements